src/Microsoft.AspNetCore.Mvc.Core/RequestSizeLimitAttribute.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// A filter that specifies the request body size limit.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class RequestSizeLimitAttribute : Attribute, IRequestSizePolicy, IOrderedFilter, IResourceFilter
+    {
+        private readonly long _bytes;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="RequestSizeLimitAttribute"/>.
+        /// </summary>
+        public RequestSizeLimitAttribute(long bytes, ILoggerFactory loggerFactory)
```

Comment:
Following the pattern of `ValidateAntiForgeryToken`, I can have an attribute that creates the filter since the DI is now involved because of the `ILoggerFactory`
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.ViewFeatures/ValidateAntiForgeryTokenAttribute.cs

@rynowak Thoughts?